### PR TITLE
[Feature] Create using storage volume

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 1]
```diff
@@ -2583,6 +2583,10 @@ public static void getDdlStmt(String dbName, Table table, List<String> createTab
                         .append("\" = \"");
                 sb.append(storageProperties.get(PropertyAnalyzer.PROPERTIES_STORAGE_CACHE_TTL)).append("\"");
 
+                sb.append(StatsConstants.TABLE_PROPERTY_SEPARATOR).append(PropertyAnalyzer.PROPERTIES_STORAGE_VOLUME)
+                        .append("\" = \"");
+                sb.append(storageProperties.get(PropertyAnalyzer.PROPERTIES_STORAGE_VOLUME)).append("\"");
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:2588]: if the storage volume name is changed, is it necessary to return the latest name?

File: fe/fe-core/src/main/java/com/starrocks/server/OlapTableFactory.java [hunk 2]
```diff
@@ -195,20 +195,39 @@ public Table createTable(LocalMetastore metastore, Database db, CreateTableStmt
...
-            if ("local".equalsIgnoreCase(volume)) {
-                table = new OlapTable(tableId, tableName, baseSchema, keysType, partitionInfo, distributionInfo, indexes);
-            } else if ("default".equalsIgnoreCase(volume)) {
-                table = new LakeTable(tableId, tableName, baseSchema, keysType, partitionInfo, distributionInfo, indexes);
-                metastore.setLakeStorageInfo(table, properties);
-            } else if (!Strings.isNullOrEmpty(volume)) {
-                throw new DdlException("Unknown storage volume \"" + volume + "\"");
-            } else if (runMode == RunMode.SHARED_DATA) {
+            if (runMode == RunMode.SHARED_DATA) {
+                if (volume.equals("local")) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/OlapTableFactory.java:204]: move `local` to `StorageVolumeMgr`

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 61 / 89 (68.54%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/SharedDataStorageVolumeMgr.java|0|7|00.00%|[53, 55, 56, 57, 59, 60, 61]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|3|00.00%|[2586, 2587, 2588]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|6|12|50.00%|[143, 144, 145, 147, 202, 203]|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|15|21|71.43%|[214, 215, 217, 219, 220, 223]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|21|26|80.77%|[397, 4691, 4693, 4694, 4697]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|10|11|90.91%|[395]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/storagevolume/StorageVolume.java|3|3|100.00%|[]|

**Review by wyb — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java (hunk 1 above)
Comment (TszKitLo40) [reply fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:2588]: We can support it later by getting storage volume by id to get the latest name.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 65 / 89 (73.03%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|3|00.00%|[2586, 2587, 2588]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|6|12|50.00%|[143, 144, 145, 147, 202, 203]|
|:large_blue_circle:|com/starrocks/server/SharedDataStorageVolumeMgr.java|4|7|57.14%|[57, 60, 61]|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|15|21|71.43%|[214, 215, 217, 219, 220, 223]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|21|26|80.77%|[397, 4691, 4693, 4694, 4697]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|10|11|90.91%|[395]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/storagevolume/StorageVolume.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 65 / 89 (73.03%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|3|00.00%|[2583, 2584, 2585]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|6|12|50.00%|[143, 144, 145, 147, 202, 203]|
|:large_blue_circle:|com/starrocks/server/SharedDataStorageVolumeMgr.java|4|7|57.14%|[57, 60, 61]|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|15|21|71.43%|[212, 213, 215, 217, 218, 221]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|21|26|80.77%|[386, 4674, 4676, 4677, 4680]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|10|11|90.91%|[395]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/storagevolume/StorageVolume.java|3|3|100.00%|[]|

**Review by sduzh — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by TszKitLo40** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**